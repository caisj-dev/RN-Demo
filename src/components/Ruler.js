import React from "react";
import { StyleSheet, View, Text } from "react-native";

function Ruler({
  segmentNum,
  rulerWidth,
  segmentWidth,
  segmentSpacing,
  spacerWidth,
  minAge,
}) {
  const ageRange = new Array(segmentNum)
    .fill(null)
    .map((item, idx) => idx + minAge);

  return (
    <View style={[{ width: rulerWidth }, styles.ruler]}>
      <View style={{ width: spacerWidth }} />
      <View style={styles.row}>
        {ageRange.map((i) => {
          const isEightEnd = (i + 2) % 10 === 0;
          return (
            <View key={i} style={styles.col}>
              <View
                style={{
                  width: segmentWidth,
                  backgroundColor: "#999",
                  height: isEightEnd ? 10 : 7,
                  marginRight: i === ageRange.length - 1 ? 0 : segmentSpacing,
                  marginBottom: isEightEnd ? 0 : 3,
                }}
              />
              <View style={styles.borderLine} />
              <Text style={styles.num}>{isEightEnd && i}</Text>
            </View>
          );
        })}
      </View>
      <View style={{ width: spacerWidth }} />
    </View>
  );
}

const styles = StyleSheet.create({
  ruler: {
    alignItems: "center",
    justifyContent: "center",
    flexDirection: "row",
  },
  row: {
    flex: 1,
    flexDirection: "row",
  },
  col: {
    flex: 1,
    flexDirection: "column",
  },
  borderLine: {
    borderBottomWidth: 2,
    borderBottomColor: "#999",
  },
  num: {
    position: "relative",
    left: -6,
    width: 30,
  },
});

export default Ruler;
