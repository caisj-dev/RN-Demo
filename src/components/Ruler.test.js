import React from "react";
import renderer from "react-test-renderer";
import Ruler from "./Ruler";

describe("<Ruler />", () => {
  it("snapshot", () => {
    const tree = renderer
      .create(
        <Ruler
          segmentNum={10}
          rulerWidth={50}
          segmentWidth={10}
          segmentSpacing={5}
          spacerWidth={20}
          minAge={18}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
