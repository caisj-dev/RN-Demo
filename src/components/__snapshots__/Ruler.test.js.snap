// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ruler /> snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "width": 50,
      },
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Object {
        "width": 20,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#999",
            "height": 10,
            "marginBottom": 0,
            "marginRight": 5,
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "borderBottomColor": "#999",
            "borderBottomWidth": 2,
          }
        }
      />
      <Text
        style={
          Object {
            "left": -6,
            "position": "relative",
            "width": 30,
          }
        }
      >
        18
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#999",
            "height": 7,
            "marginBottom": 3,
            "marginRight": 5,
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "borderBottomColor": "#999",
            "borderBottomWidth": 2,
          }
        }
      />
      <Text
        style={
          Object {
            "left": -6,
            "position": "relative",
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#999",
            "height": 7,
            "marginBottom": 3,
            "marginRight": 5,
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "borderBottomColor": "#999",
            "borderBottomWidth": 2,
          }
        }
      />
      <Text
        style={
          Object {
            "left": -6,
            "position": "relative",
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#999",
            "height": 7,
            "marginBottom": 3,
            "marginRight": 5,
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "borderBottomColor": "#999",
            "borderBottomWidth": 2,
          }
        }
      />
      <Text
        style={
          Object {
            "left": -6,
            "position": "relative",
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#999",
            "height": 7,
            "marginBottom": 3,
            "marginRight": 5,
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "borderBottomColor": "#999",
            "borderBottomWidth": 2,
          }
        }
      />
      <Text
        style={
          Object {
            "left": -6,
            "position": "relative",
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#999",
            "height": 7,
            "marginBottom": 3,
            "marginRight": 5,
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "borderBottomColor": "#999",
            "borderBottomWidth": 2,
          }
        }
      />
      <Text
        style={
          Object {
            "left": -6,
            "position": "relative",
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#999",
            "height": 7,
            "marginBottom": 3,
            "marginRight": 5,
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "borderBottomColor": "#999",
            "borderBottomWidth": 2,
          }
        }
      />
      <Text
        style={
          Object {
            "left": -6,
            "position": "relative",
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#999",
            "height": 7,
            "marginBottom": 3,
            "marginRight": 5,
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "borderBottomColor": "#999",
            "borderBottomWidth": 2,
          }
        }
      />
      <Text
        style={
          Object {
            "left": -6,
            "position": "relative",
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#999",
            "height": 7,
            "marginBottom": 3,
            "marginRight": 5,
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "borderBottomColor": "#999",
            "borderBottomWidth": 2,
          }
        }
      />
      <Text
        style={
          Object {
            "left": -6,
            "position": "relative",
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#999",
            "height": 7,
            "marginBottom": 3,
            "marginRight": 5,
            "width": 10,
          }
        }
      />
      <View
        style={
          Object {
            "borderBottomColor": "#999",
            "borderBottomWidth": 2,
          }
        }
      />
      <Text
        style={
          Object {
            "left": -6,
            "position": "relative",
            "width": 30,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "width": 20,
      }
    }
  />
</View>
`;
